File: backend/app/schemas/admin_orders.py
import uuid
from datetime import datetime
from decimal import Decimal

from app.models.enums import OrderStatus, PaymentMethod, PurchaseType
from app.schemas.company import RequesterMini
from app.schemas.order import BillingDocumentOut, OrderItemOut, ShipmentOut
from pydantic import BaseModel, ConfigDict


class AdminOrderSummaryOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    status: OrderStatus
    purchase_type: PurchaseType
    company_id: uuid.UUID | None
    company_name: str | None = None
    total_amount: Decimal
    created_at: datetime
    item_count: int
    placed_by: RequesterMini


class AdminOrderDetailsOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    status: OrderStatus
    purchase_type: PurchaseType
    company_id: uuid.UUID | None
    company_name: str | None = None
    payment_method: PaymentMethod
    total_amount: Decimal
    note: str | None
    created_at: datetime
    billing_document: BillingDocumentOut
    shipment: ShipmentOut
    items: list[OrderItemOut]
    placed_by: RequesterMini

File: frontend/src/api/adminOrders.ts
import apiFetch from "./client";
import type {
  BillingDocumentOut,
  OrderItemOut,
  OrderStatus,
  PaymentMethod,
  PurchaseType,
  ShipmentOut,
} from "./order";

export interface AdminOrderPlacer {
  id: string;
  email: string;
  first_name: string;
  last_name: string;
}

export interface AdminOrderSummary {
  id: string;
  status: OrderStatus;
  purchase_type: PurchaseType;
  company_id: string | null;
  company_name: string | null;
  total_amount: string;
  created_at: string;
  item_count: number;
  placed_by: AdminOrderPlacer;
}

export interface AdminOrderDetails {
  id: string;
  status: OrderStatus;
  purchase_type: PurchaseType;
  company_id: string | null;
  company_name: string | null;
  payment_method: PaymentMethod;
  total_amount: string;
  note: string | null;
  created_at: string;
  billing_document: BillingDocumentOut;
  shipment: ShipmentOut;
  items: OrderItemOut[];
  placed_by: AdminOrderPlacer;
}

export const fetchAdminOrders = (
  status?: OrderStatus,
  signal?: AbortSignal,
) => {
  const qs = status ? `?status=${status}` : "";
  return apiFetch<AdminOrderSummary[]>(`/admin/orders${qs}`, { signal });
};

export const fetchAdminOrder = (orderId: string, signal?: AbortSignal) =>
  apiFetch<AdminOrderDetails>(`/admin/orders/${orderId}`, { signal });

export const advanceAdminOrderStatus = (orderId: string) =>
  apiFetch<AdminOrderDetails>(`/admin/orders/${orderId}/advance-status`, {
    method: "POST",
  });
